lemmy_apub/src/inbox/community_inbox.rs

```diff
@@ -120,7 +123,12 @@ async fn handle_undo_follow(
   community: Community,
   context: &LemmyContext,
 ) -> Result<HttpResponse, LemmyError> {
-  let _undo = Undo::from_any_base(activity)?.context(location_info!())?;
+  let undo = Undo::from_any_base(activity)?.context(location_info!())?;
+  verify_activity_domains_valid(&undo, user.actor_id()?, true)?;
+
+  let object = undo.object().to_owned().one().context(location_info!())?;
+  let follow = Follow::from_any_base(object)?.context(location_info!())?;
+  verify_activity_domains_valid(&follow, user.actor_id()?, false)?;
```

Comment:
Hrm... I don't understand why one should check the domain while the other shouldn't. I think the only special case we have, is an announce, which the user and community inboxes don't recieve.

Reply:
Its very useful to have the docs open for this. In the case of `Follow`, `id` and `actor` are both from the same domain, but `object` is from a different domain (otherwise the whole thing wouldnt make sense if you think about it).

https://enterprise.lemmy.ml/docs/contributing_apub_api_outline.html#follow